CLDR-18980 Tweak constructed interval formats doc (#5775)
unicode-org/cldr: docs/site/translation/date-time.md

# Date & Time

Date and time data provides information for formatting dates, times, timezones, datetime intervals. The pages below describe date and time pieces, how to write the patterns to format dates, the symbols used in the date time patterns, and time zones.

* [Date/Time Names](https://cldr.unicode.org/translation/date-time/date-time-names)
* [Date/Time Patterns](https://cldr.unicode.org/translation/date-time/date-time-patterns)
* [Date/Time Symbols](https://cldr.unicode.org/translation/date-time/date-time-symbols)
* [Time Zones and City names](https://cldr.unicode.org/translation/time-zones-and-city-names)

> ## Error/warning messages 🚨 

Author: ```suggestion
🚨
## Error/warning messages 
```

The following are some of the error/warning messages you might see, and how to handle them.

* The first part of the message lists the Code and Value. For example:
    * Code ┃ Value ┃ Message
    * GyMMMd ┃ dd MMM y G ┃ …
* The messages might not be exactly what you see on the screen,
because parts of the message are customized.
* The links to related paths (shown in italic on this page) will open up in a separate window so that you can compare side-by-side.

> 

Author: ```suggestion
* The links to related paths (shown in italic on this page) will open up in a separate window so that you can compare side-by-side.
* Note that some errors or warnings may be due to inherited times, sometimes between calendars.
```

However, some locales change the separator if there are only two placeholders instead of three.
   * Check carefully to see whether you should change the separator or change what is in the base or other related Flexible time Codes.

**Date ┃ E d/M ┃ Numeric date separator «/» in pattern conflicts with «-», the default in _Date_ — fix as per _Date/time error/warning messages_.**
   * This warning is on the other side of the coin; it is on a pattern that contains Md, and yM (thus including MEd, GyM, etc.)
   * Look at the pattern carefully to see whether you need to change it for consistency across all the related Codes.

## Numeric time separators

**Time ┃ . ┃ Numeric time separator conflicts with «:» from the base «H:mm:ss» at _Hms_**
   * The numeric time separator conflicts with what is in the pattern for Flexible times with the Code `Hms`.
   * Check carefully to see whether you should change the separator or change what is in the base.

**Time ┃ H.mm ┃ Numeric time separator «:» in pattern conflicts with «.», the default for _Date_**
   * This warning is on the other side of the coin; it is on a pattern that contains Hm, hm, ms, or ms (thus including Hms, hms, hmv, etc.)

>    * Look at the pattern carefully to see whether you need to change it for consistency across all the related Codes.

Reviewer: ```suggestion
   * This warning is on the converse of the previous one; it is on a pattern that contains Hm, hm, ms, or ms (thus including Hms, hms, hmv, etc.)
   * Look at the pattern carefully to see whether you need to change it for consistency across all of the related Codes.
```

minor:
- removed an idiom
- grammar fix

# Date & Time

Date and time data provides information for formatting dates, times, timezones, datetime intervals. The pages below describe date and time pieces, how to write the patterns to format dates, the symbols used in the date time patterns, and time zones.

* [Date/Time Names](https://cldr.unicode.org/translation/date-time/date-time-names)
* [Date/Time Patterns](https://cldr.unicode.org/translation/date-time/date-time-patterns)
* [Date/Time Symbols](https://cldr.unicode.org/translation/date-time/date-time-symbols)

> * [Time Zones and City names](https://cldr.unicode.org/translation/time-zones-and-city-names)

Reviewer 2: For future proofing, we should probably have these be relative links.

🚨
## Error/warning messages 

The following are some of the error/warning messages you might see, and how to handle them.

* The first part of the message lists the Code and Value. For example:
    * Code ┃ Value ┃ Message

>     * GyMMMd ┃ dd MMM y G ┃ …

Reviewer 2: Was this supposed to make a table? FYI, it doesn't.

<img width="553" height="162" alt="Image" src="https://github.com/user-attachments/assets/2d5d232c-b544-441e-8111-d842a7f36b55" />

Author: It isn't meant to be a table: it's meant to mirror what the vetter's see in the Dashboard.

> * The messages might not be exactly what you see on the screen,

Reviewer 2: ```suggestion
* The messages might not be exactly what you see on the screen
```

<!-- 
DONE
afternoon1 ┃ in die middag ┃ Narrow value "in die middag" shouldn't be longer than the corresponding abbreviated value "die middag" ┃

For later
morning1 ┃ du mat. ┃ Inconsistent use of periods in abbreviations for this section. ┃
afternoon1 ┃ in die middag ┃ Narrow value "in die middag" shouldn't be longer than the corresponding abbreviated value "die middag" ┃
medium ┃ d MMM. y ┃ Your pattern (d MMM. y) is probably incorrect; abbreviated month/weekday/quarter names that need a period should include it in the name, rather than adding it to the pattern. ┃
-->

### Numeric date separators

**Date ┃ \- ┃ Numeric date separator conflicts with «/» from the base «d/M/y» at _yMd_**
   * The numeric date separator should be consistent with what is in the pattern for Flexible dates with the Code yMd.
   * That is normally also the time separator for Codes that contain Md, and yM.
However, some locales change the separator if there are only two placeholders instead of three.
   * Check carefully to see whether you should change the separator or change what is in the base or other related Flexible time Codes.

> **Date ┃ E d/M ┃ Numeric date separator «/» **in pattern** conflicts with «-», the default in _Date_ — fix as per _Date/time error/warning messages_.**

Reviewer 2: This formatting doesn't seem to work. In preview it comes out to

<img width="889" height="55" alt="Image" src="https://github.com/user-attachments/assets/0b4c2a3a-1b84-4c9c-9f18-1f3a32332cd9" />

Author: will fix

> **Date ┃ E d/M ┃ Numeric date separator «/» **in pattern** conflicts with «-», the default in _Date_ — fix as per _Date/time error/warning messages_.**

Author: ```suggestion
**Date ┃ E d/M ┃ Numeric date separator «/» in pattern conflicts with «-», the default in _Date_ — fix as per _Date/time error/warning messages_.**
```

   * This warning is on the other side of the coin; it is on a pattern that contains Md, and yM (thus including MEd, GyM, etc.)
   * Look at the pattern carefully to see whether you need to change it for consistency across all the related Codes.

## Numeric time separators

**Time ┃ . ┃ Numeric time separator conflicts with «:» from the base «H:mm:ss» at _Hms_**

>    * The numeric time separator conflicst what is in the pattern for Flexible times with the Code `Hms`.

Reviewer 2: ```suggestion
   * The numeric time separator conflicts with what is in the pattern for Flexible times with the Code `Hms`.
```

   * Check carefully to see whether you should change the separator or change what is in the base.

> **Time ┃ H.mm ┃ Numeric time separator «:» **in pattern** conflicts with «.», the default for _Date_**

Reviewer 2: Similar formatting issues here.

<img width="745" height="55" alt="Image" src="https://github.com/user-attachments/assets/c8056e7e-72cb-45f3-8390-7a3371f08a32" />

> **Time ┃ H.mm ┃ Numeric time separator «:» **in pattern** conflicts with «.», the default for _Date_**

Author: ```suggestion
**Time ┃ H.mm ┃ Numeric time separator «:» in pattern conflicts with «.», the default for _Date_**
```

There is also a warning for Intervals, to help identify where there may be unintentional inconsistencies between the interval formats and the related flexible format.
The message has 4 different pieces of information.
Here is a summary, and what it means for you is below that

* «…» is a _constructed interval format_.
* _…_ is a link to the Flexible format used to construct it.
(Note: the new interval separators are also used in the construction.)
* diffs=\[…] provides details about the differences
* samples=«…» provides two examples of what this would look like:
    * the first is using the current interval format
    * the second is using the the constructed interval format.

Take an example like the following:

| Code | Available pattern | Interval Pattern | Constructed Interval Pattern |
| -- | -- | -- | -- |
| MMMMd/M | M月d日 | MMMM d – MMMM d	| M月d日～M月d日 |
| MMMMd/d | M月d日 | MMMM d–d | M月d日～d日 |

The **Interval Pattern** in each case is unexpectedly different than the **Available pattern** for "MMMMd".
To help with this, a warning is given when it is different than a _constructed interval pattern_.
That warning will provide the constructed pattern, plus status and samples.

> Now, the constructed pattern might not be right for your locale (sometimes it needs a human touch!),

Reviewer 2: ```suggestion
Now, the constructed pattern might not be right for your locale (sometimes it needs a human touch),
```

but can reveal when you can

1. fix inconsistencies with the available formats, or
2. restructure the pattern to get it shorter or clearer, or
3. fix the related flexible format used to create the constructed pattern.

So please look at the warnings for _each_ item in the Flexible Date or Time formats,
and decide which of the above is appropriate. 

NOTE: There are cases where the constructed pattern is wrong and you will need to [add or vote for the correct value][] for your locale.

> For example

Reviewer 2: This seems orphaned.

Author: ```suggestion
```

> Date and time data provides information for formatting dates, times, timezones, datetime intervals (eg, Dec 15-18), and so on.

Reviewer 3: ```suggestion
Date and time data provides information for formatting dates, times, timezones, datetime intervals. The pages below describe date and time pieces, how to write the patterns to format dates, the symbols used in the date time patterns, and time zones.
```

> See the subpages in the sidebar.

Reviewer 3: ```suggestion

* [Date/Time Names](https://cldr.unicode.org/translation/date-time/date-time-names)
* [Date/Time Patterns](https://cldr.unicode.org/translation/date-time/date-time-patterns)
* [Date/Time Symbols](https://cldr.unicode.org/translation/date-time/date-time-symbols)
* [Time Zones and City names](https://cldr.unicode.org/translation/time-zones-and-city-names)
```

> ## Error/warning messages 🆕 

Reviewer 3: ```suggestion
<a name="error-messages"></a>
## Error/warning messages 🆕 
```

The following are some of the error/warning messages you might see, and how to handle them.
The underlined portion of each message is the part that will change depending on which row you are on.
The messages might not be exactly what you see on the screen,
because parts of the message are customized
It will open up in a separate window so that you can compare side-by-side.

> ### Numeric date separators

Reviewer 3: Should we say anything about calendar inheritance and what to do if they have an error due to this being inherited to a calendar such as Gregorian YMD where the separator may be inherited but intentionally different from Gregorian?

Author: Good idea!

Author: Added line in intro. See if that works.

> 1. Numeric date separator conflicts with «/» from the base «d/M/y» at 🔗yMd🔗

Reviewer 3: I don't understand the goal of 🔗 in this doc?

Author: That is just an artifact generated by the ConsoleCheckCLDR in place of the link HTML. 

I could underline instead, which is more like what the user would see in ST. Does that work?

Reviewer 3: Makes sense in that case.

1. Numeric date separator «/» in pattern conflicts with «.», the default for 🔗Date🔗
   * This warning is on the other side of the coin; it is on a **pattern** that contains Md, and yM (thus including MEd, GyM, etc.)
   * Look at the pattern carefully to see whether you need to change it for consistency across all the related Codes.

## Numeric time separators

1. Numeric time separator conflicts with «:» from the base «H:mm:ss» at 🔗Hms🔗
   * The numeric time separator conflicst what is in the pattern for Flexible times with the Code `Hms`.
   * Check carefully to see whether you should change the separator or change what is in the base.
2. Numeric time separator «:» in pattern conflicts with «.», the default for 🔗Date🔗
   * This warning is on the other side of the coin; it is on a **pattern** that contains Hm, hm, ms, or ms (thus including Hms, hms, hmv, etc.)
   * Look at the pattern carefully to see whether you need to change it for consistency across all the related Codes.

### Interval Formats

1. Conflicts with «d–d.» from 🔗d🔗; diffs=\[fewer]; samples=«25.–26.», «25–26.»

> There is also new warning for Intervals, to help you get consistency _among_ the interval formats, and _with_ each associated flexible format.

Reviewer 3: ```suggestion
There is also a warning for Intervals, to help identify where there may be unintentional inconsistencies between the interval formats and the related flexible format.
```

Take an example like the following:

| Code | Available pattern | Interval Pattern | Constructed Interval Pattern |
| -- | -- | -- | -- |
| MMMMd/M | M月d日 | MMMM d – MMMM d	| M月d日～M月d日 |
| MMMMd/d | M月d日 | MMMM d–d | M月d日～d日 |

The **Interval Pattern** in each case is unexpectedly different than the **Available pattern** for "MMMMd".
To help with this, a warning is given when it is different than a _constructed interval pattern_.
That warning will provide the constructed pattern, plus status and samples.

Now, the constructed pattern might not be right for your locale (sometimes it needs a human touch!),
but can reveal when you can

1. fix inconsistencies with the available formats, or
2. restructure the pattern to get it shorter or clearer, or
3. fix the related flexible format used to create the constructed pattern.

So please look at the warnings for _each_ item in the Flexible Date or Time formats,
and decide which of the above is appropriate. 

> NOTE: There are cases where the constructed pattern will just be wrong.

Reviewer 3: ```suggestion
NOTE: There are cases where the constructed pattern is wrong and you will need to [add or vote for the correct value][] for your locale.
```

| other | two fields have the same type, numeric status, and length; eg L vs M |

### Flexible Formats
   
1. date-SHORT → “dd.MM.yy” doesn't match the corresponding flexible skeleton: yMd → “d. M. y”. Fix either this or the flexible pattern
   * The "stock" date 
2. “E, d. M. y G” ⊅ “d. M. y GGGGG”: the pattern for GyMEd should contain the pattern at 🔗GyMd🔗
   * There are 5 fields (G, E, B, a, and v) that are normally prefixed or suffixed to a core pattern.
   * For example, in the majority of locales, if a Code like yMMM has a pattern «MMM y»,
then a Code like GyMMM will have a pattern like «MMM y G» or «G MMM y». That is, the literals in the pattern (the spaces in this example) may vary by locale, but the order from the core is retained.
   * Look carefully at the two patterns in the two different rows to see if it would be better (or at least as good) to change the order of one or the other.
3. Unexpected order of era/year. Expected era then year, but got field between era and year in «G M/y» for gregorian/GyM — see 🔗Info Hub🔗
   * In most locales, the era placeholder (G) is adjacent to the year placeholder (y), either before or after.

>    * Your locale might be an exceptional one, but check carefully to see whether you should move the G.

Reviewer 3: ```suggestion
   * Your locale might be an exceptional one, but check carefully to see whether you should move the G.
  
  [add or vote for the correct value]: /translation/getting-started/guide#how-to-vote 
```
